File: data/problems/LongestSubstringWithoutRepeatingCharacters/cpp/solution_expected.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int size = s.length();
        int maxLength = 0;
        unordered_map<char, int> lastVisited;

        for (int j = 0, i = 0; j < size; j++){
            if(lastVisited[s[j]] > 0) {
                i = max(lastVisited[s[j]], i);
            }
            maxLength = max(maxLength, j - i + 1);
            lastVisited[s[j]] = j + 1;
        }
        return maxLength;
    }
};

File: data/problems/LongestSubstringWithoutRepeatingCharacters/python/solution_expected.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        max_length = 0
        last_visited = {}
        left = 0

        for right, char in enumerate(s):
            if char in last_visited:
                left = max(last_visited[char], left)
            max_length = max(max_length, right - left + 1)
            last_visited[char] = right + 1

        return max_length
